### PR TITLE
initial-version

### PR DESCRIPTION
**Pull request by zacmos**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,52 +1,67 @@
...
+const { Reshuffle } = require('reshuffle')
+const { MySQLConnector } = require('reshuffle-mysql-connector')
+
+  const app = new Reshuffle()
+  const mysql = new MySQLConnector(app)
+  const connection = mysql.sdk().createConnection({
+    host: 'DB_HOST',
+    user: 'DB_USERNAME',
+    password: 'DB_PASSWORD',
+    database: 'DB_NAME'
```
Comment (foundingnimo) [inline README.md]: I think with examples we try and stick to using `process.env.DB_HOST` etc?

File: README.md [hunk 2]
```diff
@@ -1,52 +1,67 @@
...
-interface _CONNECTOR_NAME_ConnectorConfigOptions {
-  foo: string // foo description
-  bar?: number // bar description
-}
 ```
 
-#### Connector events
+#### Table of Contents
+
+_Connector actions_:
```
Comment (foundingnimo) [inline README.md:40]: We should start with the configuration here

File: src/index.ts [hunk 3]
```diff
@@ -1,61 +1,25 @@
...
-    return event
+import mysql2 from 'mysql2'
+import mysqlPromise from 'mysql2/promise'
+import { Reshuffle, BaseConnector } from 'reshuffle-base-connector'
+
+export default class MySQLConnector extends BaseConnector<null, null> {
+  private client?: any
+  private clientPromise?: any
+
+  constructor(app: Reshuffle, id?: string) {
```
Comment (foundingnimo) [inline src/index.ts]: Normally in the connector, we encapsulate the options. I suppose in this case these will be the connection details (host, db, username, password, maybe port as well?)
Are we likely to expose an sdk that lets the user create multiple connections to multiple databases, or are we doing one connector instance per db?

**Review by foundingnimo — CHANGES_REQUESTED** [review]

File: README.md [hunk 4]
```diff
@@ -20,48 +20,187 @@ const { Reshuffle } = require('reshuffle')
...
+  "SELECT age FROM users WHERE firstName = ? and lastName = ?",
+  ['John', 'Smith']
+)
 ```
 
-More details and code samples about Node MySQL 2 Client can be found [here](https://www.npmjs.com/package/mysql2)
+This action returns an object with the results of the query, where
+`fields` is an array of all field names, as returned by the query.
+Field names in a `SELECT` query are column names, or are specified
+with an `AS` clause. Every element of `rows` is uses the names in
```
Comment (foundingnimo) [inline README.md]: Remove `is` perhaps?

File: README.md [hunk 5]
```diff
@@ -20,48 +20,187 @@ const { Reshuffle } = require('reshuffle')
...
 
-##### <a name="sdkPromise"></a>Full access to the MySQL Client SDK with support of Promise API
+##### <a name="sequence"></a>Sequence action
 
 
 ```js
-const sdkPromise = mysql.sdkPromise()
+const res = await mysql.sequence(async (query) => {
+  await query("INSERT INTO users VALUES ('Miles', 'Davis', 43)")
+  return query("SELECT COUNT(*) FROM users where 1")
```
Comment (foundingnimo) [inline README.md]: is `where 1` necessary?

File: README.md [hunk 6]
```diff
@@ -20,48 +20,187 @@ const { Reshuffle } = require('reshuffle')
...
+  if (100 <= userCount) {
+    throw new Error('Too many users:', userCount)
+  }
+  return query("INSERT INTO users VALUES ('Charlie', 'Parker', 49)")
+})
+```
+
+Use `transaction` to run multiple queries as an atomic SQL transaction.
+The interface is identical to the [sequence action](#sequence), but all
+the queries issued `seq` either success or fail together. If any of the
```
Comment (foundingnimo) [inline README.md:159]: I think the `seq` here is redundant?

File: README.md [hunk 7]
```diff
@@ -20,48 +20,187 @@ const { Reshuffle } = require('reshuffle')
...
+queries fail, all queries are rolled back and an error is thrown.
+
+Consider, for example, the following code for updating a bank account
+balance:
+
+```js
+  const accountId = 289
+  const change = +1000
```
Comment (foundingnimo) [inline README.md]: Why `+1000` ?

File: src/index.ts [hunk 8]
```diff
@@ -1,24 +1,69 @@
...
+  public async query(sql: string, params?: any[]): Promise<any> {
+    try {
+      const [rows, fields] = await this.pool.query(sql, params)
+      return {
+        fields: fields,
+        rows: rows,
+        rowCount: rows ? Object.keys(rows).length : 0,
+      }
+    } catch (error) {
+      console.log(error)
```
Comment (foundingnimo) [inline src/index.ts]: This should be `app.getLogger()`

One more `console.log` below.

**Review by foundingnimo — CHANGES_REQUESTED** [review]
A few changes please. When done - you can merge and publish :)

File: README.md (hunk 6 above)
Comment (zacmos) [reply README.md:159]: It's an explanation about the differences between the transaction and sequence actions. The same description is in pgsql connector.

**Pull request merged**